compose/cli/main.py

```diff
@@ -655,20 +655,19 @@ def remove_container(force=False):
         if options['--rm']:
             project.client.remove_container(container.id, force=True)
 
-    def force_shutdown(signal, frame):
-        project.client.kill(container.id)
-        remove_container(force=True)
-        sys.exit(2)
-
-    def shutdown(signal, frame):
-        set_signal_handler(force_shutdown)
-        project.client.stop(container.id)
-        remove_container()
-        sys.exit(1)
+    signals.set_signal_handler_to_shutdown()
+    try:
+        dockerpty.start(project.client, container.id, interactive=not options['-T'])
+        exit_code = container.wait()
```

Comment:
> For example, if the second signal fires before set_signal_handler(force_shutdown) is set, there isn't much we can do about that.
Yes, but at least in that scenario the user can Control-C again. In the scenario I mentioned above, the user doesn't have a second chance.

I'm not sure if it's possible with this approach to handle all of the essential cases cleanly. I proposed an alternative approach in the issue discussion earlier today, but it would take more work. I would favor any approach that moves closer to some of the ideas I mentioned. I think it would make subsequent changes easier to do and think about, etc. With this type of exception handling, it is not so easy, and the logic is repeated twice (and would need to be more times if additional operations are added).